vaadin-grid-flow-parent/vaadin-grid-flow/src/main/resources/META-INF/resources/frontend/gridConnector.js

```diff
@@ -215,28 +207,29 @@ import { isFocusable } from '@vaadin/grid/src/vaadin-grid-active-item-mixin.js';
         });
         grid._createPropertyObserver('activeItem', '__activeItemChangedDetails', true);
 
+        grid.$connector._getPageForIndex = tryCatchWrapper(function (index) {
+          return Math.floor(index / grid.pageSize);
+        });
+
         grid.$connector._getSameLevelPage = tryCatchWrapper(function (parentKey, currentCache, currentCacheItemIndex) {
           const currentParentKey = currentCache.parentItem ? grid.getItemId(currentCache.parentItem) : root;
           if (currentParentKey === parentKey) {
             // Level match found
-            return grid._getPageForIndex(currentCacheItemIndex);
+            return grid.$connector._getPageForIndex(currentCacheItemIndex);
```

Comment:
nit: Could be inlined to `Math.floor(currentCacheItemIndex / grid.pageSize)` instead of adding the `_getPageForIndex` function.

Reply:
Done.